File: simple-stack-example-mvp-view/src/main/java/com/zhuinden/simplestackdemoexamplemvp/application/injection/NavigationModule.java
package com.zhuinden.simplestackdemoexamplemvp.application.injection;

import com.zhuinden.simplestack.Backstack;
import com.zhuinden.simplestackdemoexamplemvp.util.BackstackHolder;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Owner on 2017. 01. 27..
 */

@Module
public class NavigationModule {
    @Provides
    public Backstack backstack(BackstackHolder backstackHolder) {
        return backstackHolder.getBackstack();
    }
}

File: simple-stack-example-mvp-view/src/main/java/com/zhuinden/simplestackdemoexamplemvp/application/injection/NavigationModule.kt
package com.zhuinden.simplestackdemoexamplemvp.application.injection

import com.zhuinden.simplestack.Backstack
import com.zhuinden.simplestackdemoexamplemvp.util.BackstackHolder

import dagger.Module
import dagger.Provides

/**
 * Created by Owner on 2017. 01. 27..
 */

@Module
object NavigationModule {
    @Provides
    @JvmStatic
    fun backstack(backstackHolder: BackstackHolder): Backstack =
        backstackHolder.backstack
}
